python/pyarrow/table.pxi

```diff
@@ -77,6 +77,49 @@ cdef class ChunkedArray:
         self._check_nullptr()
         return self.chunked_array.null_count()
 
+    def __getitem__(self, key):
+        cdef int64_t item, i
+        self._check_nullptr()
+        if isinstance(key, slice):
+            return _normalize_slice(self, key)
+        else:
+            item = int(key)
```

Comment:
You don't want to allow non-integer types such as float here.

Reply:
Changed it